pmd-java/etc/grammar/Java.jjt

```diff
@@ -1721,6 +1754,25 @@ void EqualityExpression() #EqualityExpression(>1):
   InstanceOfExpression()  ( LOOKAHEAD(2) ( "==" {jjtThis.setImage("==");} | "!=" {jjtThis.setImage("!=");} ) InstanceOfExpression()  )*
 }
 
+void Pattern() #void:
+{}
+{
+    PrimaryPattern() [ GuardedPatternCondition() #GuardedPattern(2) {checkForGuardedPatterns();} ]
+}
+
+void GuardedPatternCondition() #void:
+{}
+{
+    "&&" ConditionalAndExpression()
+}
+
+void PrimaryPattern() #void:
+{}
+{
+    TypePattern()
+    | "(" Pattern() ")"
```

Comment:
I've not created a common base class yet - see AstImplUtil . Maybe we can add this later - I consider this an implementation detail. The attribute has the same name as in pmd 7 (ParenthesisDepth) and is available for all patterns.
Since we use the usual ConditionalAnd production for the guarded pattern, we don't see all the parens in the expression, but I think, we can skip this for pmd 6.